fix: booker atom failing to find event

hackice20 opened the pull request:
## What does this PR do?

This PR fixes a critical bug in the Booker Atom API where events would fail to load when an invalid `orgId` parameter was provided. The fix ensures graceful handling of non-existent organization IDs and implements proper fallback mechanisms.

- Fixes #23758

## Visual Demo (For contributors especially)

#### Video Demo:

**Before Fix:**
- API endpoint: `GET /v2/atoms/event-types/sauna/public?isTeamEvent=true&teamId=95343&username=bookings-cmfe6sk9c000dpc1rtx8k67rr-matterlife-co&orgId=97356`
- Result: **404 Not Found** error
- Booker component fails to load the event

**After Fix:**
- Same API endpoint now returns **200 OK** with event data
- Booker component successfully loads the event
- Graceful fallback when orgId is invalid

#### Reproduction Steps:
1. Navigate to a booking page with the Booker component
2. Check Network tab for API calls to `/v2/atoms/event-types/sauna/public`
3. Verify 200 OK response instead of 404 error

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox. **N/A - No documentation changes required**
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works. **Manual testing completed, unit tests can be added in follow-up**

## How should this be tested?

### Environment Setup:
- No special environment variables required
- Test against the production API endpoint

### Test Data:
- Event slug: `sauna`
- Team ID: `95343`
- Username: `bookings-cmfe6sk9c000dpc1rtx8k67rr-matterlife-co`
- Invalid orgId: `97356` (non-existent organization)

### Test Cases:

**1. Original Failing Case:**
```bash
curl "https://api.cal.com/v2/atoms/event-types/sauna/public?isTeamEvent=true&teamId=95343&username=bookings-cmfe6sk9c000dpc1rtx8k67rr-matterlife-co&orgId=97356"
```
- **Expected**: 200 OK with event data (was 404 before fix)

**2. Valid Organization ID:**
```bash
curl "https://api.cal.com/v2/atoms/event-types/sauna/public?isTeamEvent=true&teamId=95343&username=bookings-cmfe6sk9c000dpc1rtx8k67rr-matterlife-co&orgId=<valid-org-id>"
```
- **Expected**: 200 OK with event data

**3. No Organization ID:**
```bash
curl "https://api.cal.com/v2/atoms/event-types/sauna/public?isTeamEvent=true&teamId=95343&username=bookings-cmfe6sk9c000dpc1rtx8k67rr-matterlife-co"
```
- **Expected**: 200 OK with event data

### Manual Testing:
1. Open browser dev tools (F12)
2. Navigate to a booking page with Booker component
3. Check Network tab for API calls
4. Verify successful 200 responses instead of 404 errors

## Checklist

- [x] I have read the [contributing guide](https://github.com/cal.com/cal.com/blob/main/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings

## Technical Details

### Root Cause:
1. **Prisma Query Bug**: Using `findUnique` with non-unique fields `(id, isOrganization)`
2. **Non-existent Organization**: orgId 97356 didn't correspond to an existing organization
3. **No Graceful Fallback**: API would fail instead of continuing without organization context

### Solution:
1. **Fixed Prisma Query**: Changed `findUnique` to `findFirst` for proper filtering
2. **Added Organization Validation**: New `getOrganizationSlug()` method with error handling
3. **Graceful Fallback**: When orgId is invalid, API continues without organization context
4. **Team-Organization Validation**: Verifies teams actually belong to provided organization

### Files Changed:
- `apps/api/v2/src/modules/atoms/services/event-types-atom.service.ts`

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=23818) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=23818) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=23818) it.</sub>

hackice20 committed:
Booker Atom failing to find event #23758 fix

hackice20 committed:
fix: handle non-existent orgId in Booker Atom API

- Add getOrganizationSlug() method with proper findFirst query
- Handle non-existent organization IDs gracefully
- Validate team-organization relationships
- Fall back to finding events without org context when orgId is invalid

Fixes #23758

hackice20 committed:
Merge branch 'main' into fix/Booker-Atom-failing-to-find-event

hackice20 commented:
@ keithwillcode
